src/utils/waitForIncomingFunds.ts

```
@@ -0,0 +1,14 @@
+
+export async function waitForIncomingFunds(
+    wallet: Wallet
+): Promise<Coin[] | VirtualCoin[]> {
+    return new Promise(async (resolve) => {
+        const stopFunc = await wallet.notifyIncomingFunds(
+            (coins: Coin[] | VirtualCoin[]) => {
+                resolve(coins);
+                stopFunc();
+            }
+        );
+    });
```

Biome (1.9.4)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)